generators/generator-base-private.js

```diff
@@ -846,6 +859,27 @@ module.exports = class extends Generator {
         });
     }
 
+    /**
+     * Check if the generator specified as blueprint has a version compatible with current JHipster.
+     * @param {string} blueprintPkgName - generator name
+     */
+    checkJHipsterBlueprintVersion(blueprintPkgName) {
+        const blueprintPackageJson = this.findBlueprintPackageJson(blueprintPkgName);
+        if (!blueprintPackageJson || !blueprintPackageJson.dependencies || !blueprintPackageJson.dependencies['generator-jhipster']) {
+            this.warning(`Could not retrieve version of JHipster declared by blueprint '${blueprintPkgName}'`);
+            return;
+        }
+        const mainGeneratorJhipsterVersion = packagejs.version;
+        const blueprintJhipsterVersion = blueprintPackageJson.dependencies['generator-jhipster'];
+        if (mainGeneratorJhipsterVersion !== blueprintJhipsterVersion) {
```

Comment:
I'd prefer to put a semver comparison here.

I do the version check on my blueprint.

6.3.0 compatibility:
https://github.com/mshima/generator-jhipster-multitenancy2/blob/master/generators/client/patcher/files.v6.3.0.js
6.3.1 compatibility:
https://github.com/mshima/generator-jhipster-multitenancy2/blob/master/generators/client/patcher/files.v6.3.1.js

Reply:
I don't know if that makes a difference, but I will try to move generator-jhipster to peer dependency.

I install a jhipster version, but my blueprint will always bring the last version, and that version is used.

Reply:
> 
> 
> I'd prefer to put a semver comparison here.
> 

As explained above, it doesn't work. It should be the exact same version, since the blueprint generator extends the generator class from generator-jhipster. Otherwise your application is generated with a mix of different JHipster versions as explained in the issue #10415

I understand this might be annoying for your blueprint, but its behavior seems really different from usual blueprints (such as official ones and those generated by generator-jhipster-blueprint), and I think we should target those blueprints.
You still can bypass the check with the `--skip-check` flag

Reply:
> 
> I install a jhipster version, but my blueprint will always bring the last version, and that version is used.

It could roughly work if your blueprint overrides all subgenerators. Roughly, because not all subgenerators can be overridden (e.g. `app` one, and all `ci-cd` and cloud deployments ones), so you still mix different JHipster version at runtime.

Reply:
What about add kind of plugin support.

Reply:
https://github.com/jhipster/generator-jhipster/pull/10550

Reply:
Ok. Once this is merged I will change to plugin support to make a semver test against the peer dependency version.